src/libmongoc/tests/test-mongoc-exhaust.c

```diff
@@ -160,7 +156,7 @@ test_exhaust_cursor (bool pooled)
       }
       BSON_ASSERT (r);
       BSON_ASSERT (doc);
-      ASSERT_CMPINT64 (timestamp1, <, get_timestamp (client, cursor2));
+      ASSERT_CMPINT64 (generation1, ==, get_generation (client, cursor2));
```

Comment:
Why did the previous timestamp comparison use `<` but we do an equality check here?

Reply:
This is an important distinction. Closing the socket for an exhaust cursor does not mean all connections to that server should be invalidated. So we should not have cleared the pool.

I added another check to assert the next cursor is using a different stream.

Reply:
Welp... that new check didn't work out. My hypothesis is that a newly allocated stream could reuse the address. I changed this to use a separate client to check the connection count from serverStatus.

This is also what the stepdown tests do:
https://github.com/mongodb/specifications/blob/master/source/connections-survive-step-down/tests/README.rst#how-to-verify-connection-count